gno.land/pkg/gnoweb/url.go

```diff
@@ -0,0 +1,134 @@
+package gnoweb
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"regexp"
+	"strings"
+)
+
+const (
+	KindUser  PathKind = "u"
+	KindRealm PathKind = "r"
+	KindPure  PathKind = "p"
+)
+
+type GnoURL struct {
+	Kind     PathKind
+	Path     string
+	Args     string
+	WebQuery url.Values
+	Query    url.Values
+	Host     string
```

Comment:
I think `domain` would be the better name here, matching what we have currently in gno.land (ie `ChainDomain`).